lib/pure/strutils.nim

```diff
@@ -324,6 +324,18 @@ proc toOctal*(c: char): string {.noSideEffect, rtl, extern: "nsuToOctal".} =
     result[i] = chr(val mod 8 + ord('0'))
     val = val div 8
 
+proc isEmpty*(s: string): bool {.noSideEffect, procvar, rtl, extern: "nsuIsEmpty".}=
+  ## Checks whether the string `s` is empty or not.
+  ##
+  ## In this function whitespaces are stripped from `s`
+  if s.len() == 0:
+    return true
+
+  if s == nil or s == "" or s.strip() == "":
```

Comment:
In C# we have two versions: [`string.IsNullorEmpty()`](https://msdn.microsoft.com/en-us/library/system.string.isnullorempty(v=vs.110).aspx) and [`string.IsNullorWhiteSpace()`](https://msdn.microsoft.com/en-us/library/system.string.isnullorwhitespace(v=vs.110).aspx). I quite like that way of naming personally.